docs/developer-docs/web-apps/verifiable-credentials/relying-party.mdx

```diff
@@ -0,0 +1,213 @@
+
+1. Load an identity provider such as Internet Identity: Open an identity provider that will act as the communication bridge between the relying party and the issuer.
+
+2. Request credentials: Send a JSON RPC request to the issuer API for a user's credentials.
+
+3. Receive a response: Receive either a success or error message from the identity provider.
+
+### Step 1: Load the identity provider in a new window
+
+First, your canister must initiate and load an identity provider (in this example, Internet Identity) in a new window with the URL path `/vc-flow`. Opening this window will trigger the identity provider to load and send a notification to the [`window.postMessage()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) interface with the following JSON-RPC content:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "vc-flow-ready"
+}
+```
+
+The user will interact with this window to authenticate with their Internet Identity. If this window is closed by the user or times out, the request for credentials will be closed, and an error will be returned to the relying party.
+
+### Step 2: Request a verifiable credential
+
+Once the user has logged into Internet Identity and the service is ready, the relying party service can request a verifiable credential for the user by sending a JSON RPC request to the issuer API. The user must be logged into Internet Identity for the request to be successful. Below is an example of a JSON RPC request to request a credential:
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "request_credential",
+  "params": {
+    "issuer": {
+        "origin": "https://employment-info.com",
+        "canisterId": "rwlgt-iiaaa-aaaaa-aaaaa-cai"
+    },
+    "credentialSpec": {
+        "credentialType": "VerifiedEmployee",
+        "arguments": {
+          "employerName": "XYZ Ltd."
+        }
+    },
+    "credentialSubject": "2mdal-aedsb-hlpnv-qu3zl-ae6on-72bt5-fwha5-xzs74-5dkaz-dfywi-aqe"
+  }
+}
+```
+
+In this request, the following parameters must be set:
+
+- `method`: `request_credential`: Uses the method for requesting a verifiable credential.
+
+- `params`: Configuration parameters to identify which credential is being requested.
+
+    - `issuer`: The information to identify which issuer API instance the request is sent to.
+
+        - `origin`: The origin URL of the issuer.
+
+        - `canisterId`: The canister ID of the issuer API. This field is optional.
+
+  - `credentialSpec`: The specification of the credential being requested.
+
+    - `credentialType`: The type of the requested credential.
+
+    - `arguments`: Key value pairs that are specific to the credential. This field is optional.
+
+  - `credentialSubject`: The principal of the user requesting the credential.
+
+  - `derivationOrigin`: An origin that should be used for principal derivation as an alternative to the client origin. This field is optional.
```

Comment:
Again, the concept of derivationOrigin is a bit difficult to grasp. Especially in this scenario where three principals for the user is involved: 
1. The principal interacting with the RP
2. The principal the user used when interacting with the issuer to create the credential in the first place
3. The id alias used by II when requesting the credential from the issuer

Every usage of ambiguous terms makes the text more difficult to understand. In this senctence there are three levels of ambiguity 😂

`An origin that should be used for principal derivation as an alternative to the client origin. This field is optional.`

What origin, which principal, what client?

Reply:
Good point.

@jessiemongeon1 @kristoferlund how about we change to something like:

> `derivationOrigin`: An origin that should be used for the relying party's principal derivation as an alternative to the frontend hostname. This should match the `derivationOrigin` used when logging in the relying party.

Reply:
Revised per feedback